### PR TITLE
audit: Use `to_i`during assignment to prevent NPE

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
https://github.com/Homebrew/homebrew-core/pull/31514#issuecomment-416292746

### CODE REVIEW

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -130,7 +130,7 @@ def audit
       formula_count += 1
       problem_count += fa.problems.size
       problem_lines = format_problem_lines(fa.problems)
-      corrected_problem_count = options[:style_offenses]&.count(&:corrected?)
+      corrected_problem_count ||= options[:style_offenses]&.count(&:corrected?)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/audit.rb]: Maybe this should always be set to an integer of some sort?

Comment (GauthamGoli) [reply Library/Homebrew/dev-cmd/audit.rb]: https://github.com/Homebrew/brew/pull/4765/files#diff-bd0ba063e0f8aeb1f4a3997d34914c0cR77

It is being initialized as 0. Without this conditional assignment, its value is being changed to `nil`

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/audit.rb]: `0` isn't falsey so this means it'll never be set in this case. Instead use a different variable which can be the `count` or `nil` and use `.to_i` which will convert `nil` to `0.

Comment (GauthamGoli) [reply Library/Homebrew/dev-cmd/audit.rb]: Fixed.

File: Library/Homebrew/dev-cmd/audit.rb [hunk 2]
```diff
@@ -130,7 +130,7 @@ def audit
       formula_count += 1
       problem_count += fa.problems.size
       problem_lines = format_problem_lines(fa.problems)
-      corrected_problem_count = options[:style_offenses]&.count(&:corrected?)
+      corrected_problem_count = options[:style_offenses]&.count(&:corrected?).to_i
```
Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/audit.rb]: How about `options.fetch(:style_offenses, []).count(&:corrected?)`?

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/audit.rb]: I like that, yeh 👍

Comment (GauthamGoli) [reply Library/Homebrew/dev-cmd/audit.rb]: Updated. Reads much better, thanks!

**Comment by MikeMcQuaid** [comment]
Nice work @GauthamGoli :shipit:

**Comment by MikeMcQuaid** [comment]
> 0% of diff hit (target 71.15%)

Doesn't need to be this PR but this might be a good thing to address for future.

**Pull request merged**